osu.Game.Rulesets.Osu/Objects/Spinner.cs

```diff
@@ -93,6 +98,16 @@ protected override void CreateNestedHitObjects(CancellationToken cancellationTok
                     ? new SpinnerTick { StartTime = startTime, SpinnerDuration = Duration }
                     : new SpinnerBonusTick { StartTime = startTime, SpinnerDuration = Duration, Samples = new[] { CreateHitSampleInfo("spinnerbonus") } });
             }
+
+            // Add ticks that give HP across the whole spinner duration
+            int maxSpinsForHp = (int)(hp_ticks_per_minute * Duration / 60000);
+
+            for (int i = 0; i < maxSpinsForHp; i++)
+            {
+                double startTime = StartTime + (float)(i + 1) / maxSpinsForHp * Duration;
+
+                AddNested(new SpinnerHealthTick { StartTime = startTime, SpinnerDuration = Duration });
+            }
```

Comment:
Why does this spawn ticks across the entire duration of the spinner, rather than just enough to be left over for after the bonus spins are judged?

Reply:
Should be adjusted.